src/model/src/phot.F

```diff
@@ -1193,127 +1193,135 @@ END SUBROUTINE CONVCLD_PROP_ACM
+                                     XCAN(2) = Met_Data%FCH(COL,ROW )*0.35
+                                     YCAN(2) = RJ_CORR_C3R( COL,ROW)
+                                     XCAN(1) = Met_Data%FCH(COL,ROW )*0.20
+                                     YCAN(1) = RJ_CORR_C4R( COL,ROW)
                                      XCANOUT = ZCAN
                                      ZCANX(COUNTCAN) = ZCAN
-                                     RJ_CORRX (COUNTCAN) = interp_linear1_internal(XCAN,YCAN,XCANOUT)
-                                   ELSE IF ( ZCAN .LE. Met_Data%FCH( COL,ROW )*0.20 ) THEN
+                                     RJ_CORRX (COUNTCAN) =interp_linear1_internal(XCAN,YCAN,XCANOUT)
+                                   ELSE IF ( ZCAN .LE.Met_Data%FCH( COL,ROW )*0.20 ) THEN
                                      COUNTCAN = COUNTCAN + 1
-                                     XCAN(2) = Met_Data%FCH( COL,ROW )*0.20
-                                     YCAN(2) = RJ_CORR_C4R( COL,ROW )
+                                     XCAN(2) = Met_Data%FCH(COL,ROW )*0.20
+                                     YCAN(2) = RJ_CORR_C4R( COL,ROW)
                                      XCAN(1) = 0.5
-                                     YCAN(1) = RJ_CORR_BOT( COL,ROW )
+                                     YCAN(1) = RJ_CORR_BOT( COL,ROW)
                                      XCANOUT = ZCAN
                                      ZCANX(COUNTCAN) = ZCAN
-                                     RJ_CORRX (COUNTCAN) = interp_linear1_internal(XCAN,YCAN,XCANOUT)
+                                     RJ_CORRX (COUNTCAN) =interp_linear1_internal(XCAN,YCAN,XCANOUT)
                                    END IF
                                    ZCAN = ZCAN-0.5  !step down in-canopy resolution of 0.5 m
-!                                WRITE(*,*) 'Met_Data%FCH = ', Met_Data%FCH( COL,ROW ),
-!     &                                     'ZCANX (COUNTCAN) = ', ZCANX (COUNTCAN),
-!     &                                     'RJ_CORRX (COUNTCAN) = ', RJ_CORRX (COUNTCAN)
                                 END DO !end loop on canopy layers
 
+
 !Integrate to get best attenuation value to use within canopy
-                                RJ_CORR( COL,ROW ) = IntegrateTrapezoid(ZCANX(COUNTCAN:1:-1),RJ_CORRX(COUNTCAN:1:-1)) /
+                                RJ_CORR( COL,ROW ) =IntegrateTrapezoid(ZCANX(COUNTCAN:1:-1),RJ_CORRX(COUNTCAN:1:-1)) /
      &                                               ZFL
-!                                WRITE(*,*) 'RJ_CORRX      = ', RJ_CORRX(COUNTCAN:1:-1),
-!     &                                     'ZCANX         = ', ZCANX(COUNTCAN:1:-1),
-!     &                                     'RJ_CORR (int) = ', RJ_CORR( COL,ROW )
 !Apply attenuation factors above and below canopy
-                                RJ( COL,ROW, 1, : ) = RJ( COL,ROW, 1, : )*RJ_CORR( COL,ROW )
+                                RJ( COL,ROW, LEV, : ) = RJ(COL,ROW, LEV, : )*RJ_CORR( COL,ROW )
```

Comment:
The loop variable "LEV" covers all model layers

DO LEV = 1, NLAYS

You should not apply the canopy attenuation to stratosphere

Reply:
we discussed in meeting today, using LEV is correct for now